packages/lesmis-server/src/auth/attachSession.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { Request, Response, NextFunction } from 'express';
+import { SessionType } from '@tupaia/server-boilerplate';
+import { UnauthenticatedError } from '@tupaia/utils';
+
+export const attachSession = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const sessionId = req.sessionCookie?.id;
+    if (!sessionId) {
+      // within lesmis-server, not having a session means the route handlers will use the default
+      // public lesmis server user auth header
+      next();
+      return;
```

Comment:
Given that the session is optional within the context of the `lesmis-server` we should probably change the `Request` type in the global namespace to have session be optional. In `lesmis-server/src/@types/express/index.d.ts`:
```
    export interface Request {
      session?: LesmisSessionType;
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      ctx: any;
    }
```

Reply:
Good point